### PR TITLE
refactor compute_coding_progress

### PR DESCRIPTION
**Pull request by lukechurch**
@IsaackMwenda PTAL

As discussed - this is the first part of making the changes to compute_progress, separating the code for computing the coding results for a dataset for a given id into a function, so that it can be called by add as well

### CODE REVIEW

File: data_tools/compute_coding_progress.py [hunk 1]
```diff
@@ -4,39 +4,48 @@
...
+    data['coding_progress'] = {}
+
+    for id in ids:
+        data['coding_progress'][id] = compute_coding_progress(id)
+
+    data["last_update"] = strftime("%Y-%m-%d %H:%M:%S", gmtime())
+
+    print (json.dumps(data, indent=2))
 
-data["last_update"] = strftime("%Y-%m-%d %H:%M:%S", gmtime())
```
Comment (IsaackMwenda) [inline data_tools/compute_coding_progress.py]: ditto

File: data_tools/compute_coding_progress.py [hunk 2]
```diff
@@ -4,39 +4,48 @@
...
-    data['coding_progress'][id]['messages_count'] = len(messages)
-    data['coding_progress'][id]['messages_with_label'] = messages_with_labels
-    fcw.set_dataset_metrics(id, data['coding_progress'][id])
+    metrics['messages_count'] = len(messages)
+    metrics['messages_with_label'] = messages_with_labels
+    
+    # Write the metrics back if they weren't stored
+    fcw.set_dataset_metrics(id, metrics)
+    return metrics
+
```
Comment (IsaackMwenda) [inline data_tools/compute_coding_progress.py]: Remove additional line spacing

File: data_tools/compute_coding_progress.py [hunk 3]
```diff
@@ -4,39 +4,48 @@
...
-fcw.init_client(CODA_CRYPTO_TOKEN_PATH)
-data = {}
-
-ids = fcw.get_dataset_ids()
-
-data['coding_progress'] = {}
-
-for id in ids:
-    data['coding_progress'][id] = {}
+def compute_coding_progress(id):
```
Comment (IsaackMwenda) [inline data_tools/compute_coding_progress.py]: Kindly add a docstring

File: data_tools/compute_coding_progress.py (hunk 1 above)
Comment (IsaackMwenda) [inline data_tools/compute_coding_progress.py]: ditto

**Comment by lukechurch** [comment]
Thanks @IsaackMwenda

**Pull request merged**